portfolio/src/main/webapp/script.js

```diff
@@ -28,3 +28,24 @@ function addRandomFact() {
   // Add it to the page.
   document.getElementById('random-facts-container').innerText = randomFact;
 }
+
+function displayMessages() {
+  fetch('/data').then(response => response.json()).then((messageArray) => {
```

Comment:
please put each .then on a separate line, so it's more readable. Example:
fetch('/data')
  .then(...)
  .then(...)